azurerm/data_source_virtual_network_gateway_connection.go

```diff
@@ -0,0 +1,267 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"express_route_circuit_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"resource_guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ipsec_policy": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"sa_lifetime": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"sa_datasize": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"ipsec_encryption": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ipsec_integrity": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ike_encryption": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ike_integrity": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"dh_group": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"pfs_group": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"tags": tagsForDataSourceSchema(),
+		},
+	}
+}
+
+func dataSourceArmVirtualNetworkGatewayConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).vnetGatewayConnectionsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Virtual Network Gateway Connection %q (Resource Group %q) was not found", name, resGroup)
+		}
+
+		return fmt.Errorf("Error making Read request on AzureRM Virtual Network Gateway Connection %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if resp.VirtualNetworkGatewayConnectionPropertiesFormat != nil {
+		gwc := *resp.VirtualNetworkGatewayConnectionPropertiesFormat
+
+		if gwc.SharedKey != nil {
+			d.Set("shared_key", gwc.SharedKey)
+		}
+
+		if gwc.AuthorizationKey != nil {
+			d.Set("authorization_key", gwc.AuthorizationKey)
+		}
+
+		if gwc.EnableBgp != nil {
+			d.Set("enable_bgp", gwc.EnableBgp)
+		}
+
+		if gwc.IngressBytesTransferred != nil {
+			d.Set("ingress_bytes_transferred", gwc.IngressBytesTransferred)
+		}
+
+		if gwc.EgressBytesTransferred != nil {
+			d.Set("egress_bytes_transferred", gwc.EgressBytesTransferred)
+		}
+
+		if gwc.UsePolicyBasedTrafficSelectors != nil {
+			d.Set("use_policy_based_traffic_selectors", gwc.UsePolicyBasedTrafficSelectors)
+		}
+
+		if gwc.ExpressRouteGatewayBypass != nil {
+			d.Set("express_route_gateway_bypass", gwc.ExpressRouteGatewayBypass)
```

Comment:
running the tests for this, it appears this field is missing from the Schema - I hope you don't mind but so that these pass I'm going to push a commit adding this